tests/integration/suite/actors/rebalancing-double-activation/rebalancing-double-activation.go

```diff
@@ -0,0 +1,416 @@
+			daprd.WithPlacementAddresses("localhost:"+strconv.Itoa(i.place.Port())),
+			daprd.WithAppPort(i.srv[j].Port()),
+			// Daprd is super noisy in debug mode when connecting to placement.
+			daprd.WithLogLevel("info"),
+		)
+	}
+
+	return []framework.Option{
+		framework.WithProcesses(i.place, i.srv[0], i.srv[1], i.daprd[0], i.daprd[1]),
+	}
+}
+
+func (i *rebalancingDoubleActivation) Run(t *testing.T, ctx context.Context) {
+	i.place.WaitUntilRunning(t, ctx)
+
+	// Wait for daprd to be ready
+	for j := 0; j < 2; j++ {
+		i.daprd[j].WaitUntilRunning(t, ctx)
+	}
+	// Wait for actors to be ready
+	for j := 0; j < 2; j++ {
+		i.handler[j].WaitForActorsReady(ctx)
+	}
+
+	// Establish a connection to the placement service
+	placementClientReady := make(chan error)
+	placementCtx, placementCancel := context.WithCancel(ctx)
+	defer placementCancel()
+	go i.getPlacementClient(placementCtx, placementClientReady)
+	select {
+	case <-ctx.Done():
+		require.Fail(t, "Placement client not ready in time")
+	case err := <-placementClientReady:
+		require.NoError(t, err)
+	}
+
+	client := util.HTTPClient(t)
+
+	// Do a bunch of things in parallel
+	errCh := make(chan error)
+
+	// To start, monitor for double activations
+	go func() {
+		errs := make([]error, 0)
+		for doubleAct := range i.doubleActivationCh {
+			// An empty message is a signal to stop
+			if doubleAct == "" {
+				break
+			}
+			errs = append(errs, fmt.Errorf("double activation of actor %s", doubleAct))
+		}
+		if len(errs) > 0 {
+			errCh <- errors.Join(errs...)
+		} else {
+			errCh <- nil
+		}
+	}()
+
+	// Schedule reminders to be executed in 0s
+	for j := 0; j < iterations; j++ {
+		go func(j int) {
+			rctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+			body := `{"dueTime": "0s"}`
+			daprdURL := fmt.Sprintf("http://localhost:%d/v1.0/actors/myactortype/myactorid-%d/reminders/reminder%d", i.daprd[0].HTTPPort(), j, j)
+			req, rErr := http.NewRequestWithContext(rctx, http.MethodPost, daprdURL, strings.NewReader(body))
+			req.Header.Set("Content-Type", "application/json")
+			if rErr != nil {
+				errCh <- fmt.Errorf("failed scheduling reminder %d: %w", j, rErr)
+				return
+			}
+			resp, rErr := client.Do(req)
+			if rErr != nil {
+				errCh <- fmt.Errorf("failed scheduling reminder %d: %w", j, rErr)
+				return
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusNoContent {
+				rb, _ := io.ReadAll(resp.Body)
+				errCh <- fmt.Errorf("failed scheduling reminder %d: status code is %d. Body: %s", j, resp.StatusCode, string(rb))
+				return
+			}
+			errCh <- nil
+		}(j)
+	}
+
+	// In parallel, add another node to the placement which will trigger a rebalancing
+	go func() {
+		rErr := i.reportStatusToPlacement(ctx, i.placementStream, []string{"myactortype"})
+		if rErr != nil {
+			errCh <- fmt.Errorf("failed to trigger rebalancing: %w", rErr)
+		} else {
+			errCh <- nil
+		}
+	}()
+
+	// Also invoke the same actors using actor invocation
```

Comment:
Actor invocation was already doing a lookup on the placement tables before invoking: https://github.com/ItalyPaleAle/dapr/blob/0762af919d768c7de96b9deecdb0e9179c8df793/pkg/actors/actors.go#L406-L418 Reminders weren't.

So this is here:

- Because invocation always uses the "correct" address and reminders don't. If we didn't do that, we may not get the double activation
- To confirm that invocation will always do a lookup